src/services/fetchers/EvmTWAPGasPriceFetcher.ts

```diff
@@ -21,16 +21,16 @@ PolygonGasPrice-TWAP20:
           chainId: polygon
  */
 @injectable()
-class EvmTWAPGasPriceFetcher {
+class EvmTWAPGasPriceFetcher implements FeedFetcherInterface {
   @inject(BlockchainGasRepository) protected gasRepository!: BlockchainGasRepository;
   @inject(PriceDataRepository) private priceDataRepository!: PriceDataRepository;
 
   static fetcherSource = '';
 
-  async apply(
-    {twap = 20, chainId}: {twap: number; chainId: ChainsIds},
-    timestamp: number,
-  ): Promise<number | undefined> {
+  async apply(params: {twap: number; chainId: ChainsIds}, options: FeedFetcherOptions): Promise<number | undefined> {
+    const {twap = 20, chainId} = params;
+    const {timestamp = 0, base: feedBase, quote: feedQuote} = options;
```

Comment:
same issue with timestamp here, please check all places and fix, if it was required it must remain that way. we can not change logic.

Reply:
True, I think here the best solution is to make it mandatory in every place and pass a 0 to those which do not need it.

Reply:
we should not use `0` as correct value or as logic to handle required/optional field. in fetchers where it is required let's required it, that's it. simply add `if (!timestamp).. throw`

Reply:
In this case, if timestamp is required (it will be a number), that means that if we set is accidentally to -1 this wouldn't throw anything and it would continue executing. What do you think?

Reply:
timestamp is set by time service, if you want to make sure it is not `-1` this this condition should be in time service. 
nobody will set it manually to -1, but if you want this condition to be stronger sure, you can do `if (!timestamp || timestamp <= 0).. throw`. so I still stand by not setting it to any value if it is not required.

Reply:
timestamp is set by time service, if you want to make sure it is not `-1` this this condition should be in time service. 
nobody will set it manually to -1, but if you want this condition to be stronger sure, you can do `if (!timestamp || timestamp <= 0).. throw`. so I still stand by not setting it to any value if it is not required.

Reply:
correct me if I am wrong, the case `timestamp <= 0` is also true when the condition `!timestamp` is true. I think `!timestamp` is true only and only if `timestamp = 0`. Isn't it?

Reply:
https://www.typescriptlang.org/play/

Reply:
Sorry that the link brings me to a template with a non-related code. I think the link is missing something.
I imagine I am missing something in the reasoning I did, I know JS has some interesting behaviours with this kind of assersions.

![image](https://github.com/umbrella-network/pegasus/assets/25662036/a375b6b2-c8b3-4a90-9965-556f253d980c)

Reply:
this link is link to typescript playground you can use to get answer to your question about js behaviour, or you can simply write simple unit test for all this cases.